yarn-project/end-to-end/src/devnet/e2e_devnet.test.ts

```diff
@@ -0,0 +1,226 @@
+      ).protocolContractAddresses.gasToken,
+      txReceipt.wallet,
+    );
+    const balance = await feeJuice.methods.balance_of_public(l2Account.getAddress()).simulate();
+    expect(balance).toEqual(amount - txReceipt.transactionFee!);
+  });
+
+  type OptionValue = undefined | boolean | { toString(): string };
+  type ArgumentValue = { toString(): string };
+
+  function cli<T>(cliCommand: string): Promise<T>;
+  function cli<T>(cliCommand: string, args: ArgumentValue[]): Promise<T>;
+  function cli<T>(cliCommand: string, opts: Record<string, OptionValue>): Promise<T>;
+  function cli<T>(cliCommand: string, args: ArgumentValue[], opts: Record<string, OptionValue>): Promise<T>;
+  function cli<T>(
+    cliCommand: string,
+    _args?: ArgumentValue[] | Record<string, OptionValue>,
+    _opts?: Record<string, OptionValue>,
+  ): Promise<T> {
+    const { promise, resolve, reject } = promiseWithResolvers<T>();
+    const args = Array.isArray(_args) ? _args : [];
+    const opts = _args && !Array.isArray(_args) ? _args : typeof _opts !== 'undefined' ? _opts : {};
+
+    const cmdArguments = args.map(x => x.toString());
+
+    opts.json = true;
+    const cmdOptions = Object.entries(opts)
+      .filter((entry): entry is [string, { toString(): string }] => typeof entry[1] !== 'undefined')
+      .map(([key, value]) =>
+        typeof value === 'boolean' ? (value ? `--${key}` : `--no-${key}`) : `--${key} ${value.toString()}`,
+      );
+
+    const cmd = `${AZTEC_CLI} ${cliCommand} ${cmdArguments.join(' ')} ${cmdOptions.join(' ')}`;
+    logger.info(`Running aztec-cli: ${cmd}`);
+    const child = exec(cmd, {
+      cwd: tmpdir(),
+      env: {
+        NODE_OPTIONS: '--no-warnings',
+      },
+    });
+
+    let err = '';
+    child.stderr?.on('data', data => {
+      logger.error(data.toString());
+      err += data.toString();
+    });
+
+    let out = '';
+    child.stdout?.on('data', data => {
+      out += data.toString();
+    });
+
+    child.on('error', reject);
+    child.on('close', (code, signal) => {
+      if (code === 0) {
+        const res = JSON.parse(out);
+        logger.info(`aztec-cli JSON output: ${JSON.stringify(res)}`);
+        resolve(res);
+      } else {
+        reject(new Error(`aztec-cli non-zero exit: code=${code} signal=${signal} ${err}`));
+      }
+    });
+
+    return promise;
+  }
+
+  async function getL1Balance(address: string, token?: EthAddress): Promise<bigint> {
+    const { balance } = await cli<{ balance: string }>('get-l1-balance', [address], {
+      'l1-rpc-url': ETHEREUM_HOST!,
+      'l1-chain-id': l1ChainId.toString(),
+      token,
+    });
+
+    return BigInt(balance);
+  }
+
+  async function advanceChainForBridgingToWork() {
+    const { l1ChainId, protocolVersion } = await pxe.getNodeInfo();
+    const counter = await TestContract.deploy(
+      new SignerlessWallet(pxe, new DefaultMultiCallEntrypoint(l1ChainId, protocolVersion)),
+    )
+      .send({ universalDeploy: true })
```

Comment:
Forgot what's the default here, but I'd skip both class registration and instance public deployment, just to make things faster.